framework/src/interfacekernels/InterfaceKernel.C

```diff
@@ -113,28 +114,33 @@ InterfaceKernel::computeOffDiagElemNeighJacobian(Moose::DGJacobianType type, uns
 }
 
 void
-InterfaceKernel::computeOffDiagJacobian(unsigned int jvar)
+InterfaceKernel::computeElementOffDiagJacobian(unsigned int jvar)
 {
   if (jvar == _var.number())
-  {
-    computeJacobian(jvar);
-    computeOffDiagElemNeighJacobian(Moose::NeighborElement, jvar);
-  }
+    computeElemNeighJacobian(Moose::ElementElement);
   else if (jvar == _neighbor_var.number())
-  {
-    computeJacobian(jvar);
-    computeOffDiagElemNeighJacobian(Moose::ElementNeighbor, jvar);
-  }
+    computeElemNeighJacobian(Moose::ElementNeighbor);
   else
   {
     // Compute element-element Jacobian
     computeOffDiagElemNeighJacobian(Moose::ElementElement, jvar);
 
-    // Compute neighbor-element Jacobian
-    computeOffDiagElemNeighJacobian(Moose::NeighborElement, jvar);
-
     // Compute element-neighbor Jacobian
     computeOffDiagElemNeighJacobian(Moose::ElementNeighbor, jvar);
+  }
+}
+
+void
+InterfaceKernel::computeNeighborOffDiagJacobian(unsigned int jvar)
+{
+  if (jvar == _var.number())
+    computeElemNeighJacobian(Moose::NeighborElement);
+  else if (jvar == _neighbor_var.number())
```

Comment:
right. I did not consider `_var == _neighbor_var` case carefully. We will need a flag to indicate if the `jvar` is one of current and neighbor variable. Do you have any example so I can put into a meaningful test? Otherwise I will just come one.